go/pkg/chunker/chunker.go

```diff
@@ -139,65 +137,47 @@ func (c *Chunker) Next() (*Chunk, error) {
 		return nil, ErrEOF
 	}
 	c.initialized = true
-	bytesLeft := c.bytesLeft()
 	if c.digest.Size == 0 {
 		return emptyChunk, nil
 	}
-	toRead := IOBufferSize
-	if int64(toRead) > bytesLeft {
-		toRead = int(bytesLeft)
-	}
-	if c.contents != nil {
-		c.buffer = c.contents
-	} else if c.reader == nil {
+	if c.contents == nil && c.reader == nil {
```

Comment:
Comment that this handles the case where this is the first call to Next() and not after a reset?